plugins/modules/df_service.py

```diff
@@ -35,17 +36,17 @@
 options:
   env_crn:
     description: 
-        - The CRN of the CDP Environment to host the Dataflow Service
-        - Required when state is present
+      - The CRN of the CDP Environment to host the Dataflow Service
+      - The environment name can also be provided, instead of the CRN
+      - Required when state is present
     type: str
-    required: False
+    required: Conditional
```

Comment:
IIRC, this has to be a Boolean.

Reply:
Boolean does not capture complex criteria. If you use `required_if`, for example, we can't say that required is neither `False` nor `True`. I actually copy the `Conditional` example from other modules I found in here.

Reply:
I can omit this if you prefer and add in the description when it's required and when it's not.
Neither `False` nor `True` are correct in this context.

Reply:
Fixed.